### PR TITLE
NO-SNOW Fix missing ubuntu packages error in CI

### PR DESCRIPTION
**Pull request by sfc-gh-mbobowski**
# Overview

NO-SNOW

CI recently started to fail because of some ubuntu packages disappearance.

## Pre-review checklist
- [ ] This change should be part of a Behavior Change Release. See [go/behavior-change](http://go/behavior-change).
- [ ] This change has passed Merge gate tests
- [ ] Snowpipe Changes
- [ ] Snowpipe Streaming Changes
- [x] This change is TEST-ONLY
- [ ] This change is README/Javadocs only
- [ ] This change is protected by a config parameter <PARAMETER_NAME> eg `snowflake.ingestion.method`.
    - [ ] `Yes` - Added end to end and Unit Tests. 
    - [ ] `No` - Suggest why it is not param protected
- [ ] Is his change protected by parameter <PARAMETER_NAME> on the server side?
    - [ ] The parameter/feature is not yet active in production (partial rollout or PrPr, see [Changes for Unreleased Features and Fixes](http://go/ppp-prpr)).
    - [ ] If there is an issue, it can be safely mitigated by turning the parameter off. This is also verified by a test (See [go/ppp](http://go/ppp)).

### CODE REVIEW

**Review by sfc-gh-rsawicki — APPROVED** [review]

File: .github/workflows/End2EndTestApache.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ on:
 
 jobs:
   build_and_test:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
```
Comment (sfc-gh-achyzy) [inline .github/workflows/End2EndTestApache.yml:11]: That is super risky... I don't think we should do that.

Comment (sfc-gh-achyzy) [reply .github/workflows/End2EndTestApache.yml:11]: Let's use the current latest instead of the latest tag

Comment (sfc-gh-achyzy) [reply .github/workflows/End2EndTestApache.yml:11]: Besides, I think I tried to bump it some time ago, and there were issues with some of the packages.

Comment (sfc-gh-mbobowski) [reply .github/workflows/End2EndTestApache.yml:11]: Ok, let's try with 24.04 then.

Comment (sfc-gh-mbobowski) [reply .github/workflows/End2EndTestApache.yml]: 24.04 works for non-Python workflows. For e2e tests there is some package version collision in 24.04, but not in 22.04.

If CI passes I will go on with combination of 22 and 24 to unblock development.

**Pull request merged**